### PR TITLE
[Contact] Fix PHPUnit 8 compatibility issues

### PR DESCRIPTION
**Pull request by srsathish92**
### Description (*)
This PR fixes the PHPUnit 8 compatibility issues of existing unit tests for Magento_Contact module.

### Related Pull Requests
N/A

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
1. Run tests in phpunit 6 (currently in 2.4-develop) using vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/Contact/
2. Install phpunit 8, run tests using vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/Contact/

### Questions or comments
If any please let me know your feedback

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Contact/Test/Unit/Block/ContactFormTest.php [hunk 1]
```diff
@@ -3,63 +3,75 @@
...
 use Magento\Contact\Block\ContactForm;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager as ObjectManagerHelper;
+use Magento\Framework\UrlInterface;
+use Magento\Framework\View\Element\Template\Context;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
 
-class ContactFormTest extends \PHPUnit\Framework\TestCase
+/**
+ * Test Covers Magento\Contact\Block\ContactForm
```
Comment (dmytro-ch) [inline app/code/Magento/Contact/Test/Unit/Block/ContactFormTest.php]:
```suggestion
 * @covers \Magento\Contact\Block\ContactForm
```

File: app/code/Magento/Contact/Test/Unit/Controller/Index/IndexTest.php [hunk 2]
```diff
@@ -3,82 +3,102 @@
...
+     * @var ResultFactory|MockObject
      */
-    protected $resultFactory;
+    private $resultFactoryMock;
 
     /**
-     * @var \Magento\Framework\UrlInterface|\PHPUnit_Framework_MockObject_MockObject
+     * @var UrlInterface|MockObject
      */
     private $url;
```
Comment (dmytro-ch) [inline app/code/Magento/Contact/Test/Unit/Controller/Index/IndexTest.php]: Please add suffix Mock to the stub properties.
```suggestion
    private $urlMock;
```

File: app/code/Magento/Contact/Test/Unit/Controller/Index/IndexTest.php [hunk 3]
```diff
@@ -3,82 +3,102 @@
...
 use Magento\Framework\Controller\ResultFactory;
 use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager as ObjectManagerHelper;
+use Magento\Framework\UrlInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
 
-class IndexTest extends \PHPUnit\Framework\TestCase
+/**
+ * Test covers Magento\Contact\Controller\Index\Index
```
Comment (dmytro-ch) [inline app/code/Magento/Contact/Test/Unit/Controller/Index/IndexTest.php]: Please use `@covers` annotation instead
```suggestion
 * @covers \Magento\Contact\Controller\Index\Index
```

File: app/code/Magento/Contact/Test/Unit/Controller/Index/PostTest.php [hunk 4]
```diff
@@ -4,140 +4,145 @@
...
 use Magento\Framework\Controller\Result\Redirect;
+use Magento\Framework\Controller\Result\RedirectFactory;
+use Magento\Framework\Message\ManagerInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager as ObjectManagerHelper;
+use Magento\Framework\UrlInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
 
 /**
+ * Test covers Magento\Contact\Controller\Index\Post
```
Comment (dmytro-ch) [inline app/code/Magento/Contact/Test/Unit/Controller/Index/PostTest.php]:
```suggestion
 * @covers \Magento\Contact\Controller\Index\Post
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hello @srsathish92, thank you for your contribution!
Could you please fix minor CR recommendations.

Thank you!

**Comment by srsathish92** [comment]
@dmytro-ch Thanks for the feedback, I have fixed as per your recommendations, can you please review now.

**Review by dmytro-ch — DISMISSED** [review]
Hi @srsathish92, great job! Thank you for the updates.

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-7277](https://jira.corp.magento.com/browse/ENGCOM-7277) has been created to process this Pull Request
:eight_spoked_asterisk: @dmytro-ch, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Before:**
Screenshot from 2020-04-14 13-00-12

**After:**
Screenshot from 2020-04-14 13-02-50

File: app/code/Magento/Contact/Test/Unit/Controller/IndexTest.php [hunk 5]
```diff
@@ -64,12 +73,9 @@ protected function setUp()
      *
      * @expectedException \Magento\Framework\Exception\NotFoundException
```
Comment (lbajsarowicz) [inline app/code/Magento/Contact/Test/Unit/Controller/IndexTest.php]: This one should be replaced with `$this->expectException`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
QA team noticed the leftover when it comes to `@expectedException`

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7277](https://jira.corp.magento.com/browse/ENGCOM-7277) has been created to process this Pull Request

**Pull request merged**